File: ADA/programs/BubbleSort.cpp
#include <iostream>
using namespace std;

void bubbleSort(int arr[], int n) {
    for(int i = 0; i < n - 1; i++) {
        for(int j = 0; j < n - i - 1; j++) {
            if(arr[j] > arr[j+1]) {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int main() {
    int arr[] = {4, 1, 3, 5, 12, 4, 4};
    bubbleSort(arr, 7);
    for(int i = 0; i < 7; i++)
        cout << arr[i] << " ";
    cout << "\n";
}

File: Python/sorting/BubbleSort.py
# Bubble Sort
def bubbleSort(arr):
    n = len(arr)
    flag = False

    for i in range (n-1):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                flag = True
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
            if not flag:
                return

arr = [5, 3, 1, 2 , 6, 4]
bubbleSort(arr)

print(arr)
